wgpu/src/backend/wgpu_core.rs

```diff
@@ -1537,6 +1539,19 @@ impl crate::Context for Context {
         }
     }
 
+    #[cfg(webgl)]
+    fn buffer_get_mapped_range_as_array_buffer(
+        &self,
+        buffer: &Self::BufferId,
+        buffer_data: &Self::BufferData,
+        sub_range: Range<wgt::BufferAddress>,
+    ) -> js_sys::ArrayBuffer {
+        let mapped_range = self.buffer_get_mapped_range(buffer, buffer_data, sub_range);
+        let array_buffer = js_sys::Uint8Array::new_with_length(mapped_range.slice().len() as u32);
+        array_buffer.copy_from(mapped_range.slice());
+        array_buffer.buffer()
+    }
```

Comment:
I don't think this is correct - the changes to the array buffer won't be reflected in the upstream changes.

Reply:
ah you're right. We'd need a special unmap operation then! We'll have to make this function webgpu only and thus return an `Option` in the dynamic interface provided by `instance`

Reply:
addressed in [6c10575](https://github.com/gfx-rs/wgpu/pull/5044/commits/6c105754e92da2997aeb15d461815aeaec94f2da)